controllers/pkg/giteaclient/giteaclient.go

```diff
@@ -36,10 +37,25 @@ type GiteaClient interface {
 	Get() *gitea.Client
 }
 
-func New(client resource.APIPatchingApplicator) GiteaClient {
-	return &gc{
-		client: client,
+var lock = &sync.Mutex{}
+
+var singleInstance *gc
+
+func New(client resource.APIPatchingApplicator, ctx context.Context) GiteaClient {
```

Comment:
I guess my only question is whether this should be called "New" or something else. I wonder if there is a convention in Go for singletons. Otherwise LGTM (but need some unit tests).

Reply:
> I guess my only question is whether this should be called "New" or something else. I wonder if there is a convention in Go for singletons. Otherwise LGTM (but need some unit tests).

`getInstance` would be better. Unit tests may require a mocking framework per my initial analysis, but am open for suggestions

Reply:
@johnbelamaric FYI I have added some basic boundary tests as a start